### PR TITLE
[channels] Add synchronous static channel setting.

### PR DESCRIPTION
**Pull request by llyzs**
Only rdpsnd and rdpdr use this setting in this PR for now. Note that rdpsnd could use SynchronousDynamicChannel setting depending on which channel it operates upon.

### CODE REVIEW

File: channels/rdpdr/client/rdpdr_main.h [hunk 1]
```diff
@@ -70,6 +70,8 @@ typedef struct
 	DEVMAN* devman;
 	BOOL ignoreInvalidDevices;
 
+	BOOL async;
```
Comment (akallabeth) [inline channels/rdpdr/client/rdpdr_main.h:73]: public header, add at end of struct!

File: channels/rdpsnd/client/rdpsnd_main.c [hunk 2]
```diff
@@ -81,6 +81,7 @@ struct rdpsnd_plugin
 	BOOL attached;
 	BOOL connected;
 	BOOL dynamic;
+	BOOL async;
```
Comment (akallabeth) [inline channels/rdpsnd/client/rdpsnd_main.c:84]: public header, add at end of struct!

**Review by akallabeth — COMMENTED** [review]
overall LGTM, two things though:

1. new settings at end of structs
2. might be a nice thing to unify these settings (queue, thread, ...) in a common set of functions that are shared by the channels

**Comment by llyzs** [comment]
1. Fixed.
2. Sounds like quite a bit changes, probably another change set if needed.

**Review by akallabeth — APPROVED** [review]
LGTM

File: channels/rdpsnd/client/rdpsnd_main.c [hunk 3]
```diff
@@ -1636,13 +1652,17 @@ static UINT rdpsnd_on_data_received(IWTSVirtualChannelCallback* pChannelCallback
...
+		{
+			Stream_Release(copy);
+			return ERROR_INTERNAL_ERROR;
+		}
+		return CHANNEL_RC_OK;
 	}
-
-	return CHANNEL_RC_OK;
+	else
+		return rdpsnd_recv_pdu(plugin, copy);
```
Comment (mfleisz) [inline channels/rdpsnd/client/rdpsnd_main.c:1665]: Nit-picking but imo code like this:
```
if (expr)
    return something special
return default;
```
is less error prone (i.e. to future changes that might introduce a path in the `if` or `else` that do not return) than:
```
if
    return;
else
    return
```

Comment (llyzs) [reply channels/rdpsnd/client/rdpsnd_main.c]: Here I think it makes more sense if I just leave the last return as the last line and check error in the else path.

**Review by mfleisz — APPROVED** [review]
LGTM

**Pull request merged**